bonelab/cli/RapidPrototype.py

```diff
@@ -3,158 +3,978 @@
 import os
 import sys
 import time
+import math
 import vtk
 import vtkbone
+import numpy as np
 
 from bonelab.util.echo_arguments import echo_arguments
+from bonelab.util.time_stamp import message
 
-#  Utility function
-def log(msg, *additionalLines):
-   """Print message with time stamp.
-   
-   The first argument is printed with the a time stamp.
-   Subsequent arguments are printed one to a line without a timestamp.
-   """
-   print('{0:8.2f} {1:s}'.format(time.time()-start_time, msg))
-   for line in additionalLines:
-       print(" " * 9 + line)
-start_time = time.time()
-
-def CreateSTL(input_filename, output_filename, gaussian, radius, 
-              marching_cubes, decimation, visualize, overwrite):
-  
-  # Check if output exists and should overwrite
-  if os.path.isfile(output_filename) and not overwrite:
-    result = input('File \"{}\" already exists. Overwrite? [y/n]: '.format(output_filename))
+def CheckExt(choices):
+    class Act(argparse.Action):
+        def __call__(self,parser,namespace,fname,option_string=None):
+            ext = os.path.splitext(fname)[1][1:]
+            if ext not in choices:
+                option_string = '({})'.format(option_string) if option_string else ''
+                parser.error("File extension must be {}{}".format(choices,option_string))
+            else:
+                setattr(namespace,self.dest,fname)
+
+    return Act
+
+class Range(object):
+    def __init__(self, start, end):
+        self.start = start
+        self.end = end
+
+    def __eq__(self, other):
+        return self.start <= other <= self.end
+
+    def __contains__(self, item):
+        return self.__eq__(item)
+
+    def __iter__(self):
+        yield self
+
+    def __str__(self):
+        return '[{0},{1}]'.format(self.start, self.end)
+
+def printMatrix4x4(m):
+    precision = 4
+    delimiter=','
+    formatter = '{{:8.{}f}}'.format(precision)
+        
+    for i in range(4):
+      row_data = delimiter.join([formatter.format(float(m.GetElement(i,j))) for j in range(4)])
+      print('[ '+row_data+' ]')
+        
+def diagonal(x, y, z):
+    return math.sqrt(math.pow(x,2)+math.pow(y,2)+math.pow(z,2))
+
+def applyTransform(transform, polydata):
+    
+    if "None" in transform: # do nothing
+      return polydata
+    
+    try:    
+       mat4x4 = readTransform(transform)
+    except OSError:
+       print("ERROR: File could not be read: " + transform)
+       exit(0)
```

Comment:
Do you actually want to exit here? A more typical solution would be to throw an exception.
```suggestion
       raise SystemError('File could not be read: ' + transform)
```